src/api/foreground/foreground-setup-wallet-sdk.ts

```diff
@@ -12,102 +12,134 @@ export function setupWalletSDK(targetWindow: Window = window) {
   /** Init events */
   const events = mitt<InjectedEvents>();
 
-  /** Init wallet API */
-  const WalletAPI: Record<string, any> = {
-    walletName: "ArConnect",
-    walletVersion: version,
-    events
-  };
-
-  /** Inject each module */
+  /**
+   * Store modules in a map for quick lookup.
+   */
+  const moduleMap = new Map<string, (typeof foregroundModules)[number]>();
   for (const mod of foregroundModules) {
-    /** Handle foreground module and forward the result to the background */
-    WalletAPI[mod.functionName] = (...params: any[]) =>
-      new Promise<any>(async (resolve, reject) => {
-        // execute foreground module
-        // TODO: Use a default function for those that do not have/need one and see if chunking can be done automatically or if it is needed at all:
-        const foregroundResult = await mod.function(...params);
-
-        // construct data to send to the background
-        const callID = nanoid();
-        const data: ApiCall & { ext: "arconnect" } = {
-          type: `api_${mod.functionName}`,
-          ext: "arconnect",
-          callID,
-          data: {
-            params: foregroundResult || params
-          }
-        };
+    moduleMap.set(mod.functionName, mod);
+  }
 
-        // TODO: Replace `postMessage` with `isomorphicSendMessage`, which should be updated to handle
-        // chunking automatically based on data size, rather than relying on `sendChunk` to be called from
-        // the foreground scripts manually.
+  /**
+   * Utility function to handle the actual "call-foreground-then-background" flow.
+   * This can be reused by the Proxy handler below.
+   */
+  async function callForegroundThenBackground(
+    fnName: string,
+    params: any[]
+  ): Promise<any> {
+    const mod = moduleMap.get(fnName);
+
+    return new Promise<any>(async (resolve, reject) => {
+      // execute foreground module
+      // TODO: Use a default function for those that do not have/need one and see if chunking can be done automatically or if it is needed at all:
+      const foregroundResult = mod ? await mod.function(...params) : null;
+
+      // 2. Prepare the payload to send
+      const callID = nanoid();
+      const data: ApiCall & { ext: "arconnect" } = {
+        type: `api_${mod.functionName}`,
+        ext: "arconnect",
```

Comment:
This value would need to change to `"arconnectEmbedded"` based on `import.meta.env?.VITE_IS_EMBEDDED_APP`